tests/duty.test.ts

```diff
@@ -0,0 +1,170 @@
+import lodash from 'lodash';
+import request from 'supertest';
+import { Express } from 'express';
+import { Collection, ObjectId } from 'mongodb';
+
+import { getServer } from '../app';
+import { getDutiesCollection } from '../db_connection';
+import { Duty, DutySchema } from '../services/schemas/duty.zschema';
+import { createTestDuty, createTestDutyWithId } from './data/duty.data';
+
+let server: Readonly<Express>;
+let collection: Readonly<Collection<Duty>>;
+
+beforeAll(async () => {
+  collection = await getDutiesCollection();
+  server = getServer();
+});
+
+beforeEach(async () => {
+  await collection.deleteMany({});
+});
+
+afterAll(async () => {
+  collection.deleteMany({});
+});
+
+describe('POST /duties', () => {
```

Comment:
Test what happens when you insert a soldier with the same tt.

Reply:
Also, what if you send something in the right field but the wrong type ?

Reply:
resolved